lib/hawk/model/association.rb

```diff
@@ -18,47 +18,88 @@ def initialize(attributes = {}, params = {})
         super
         if attributes.size > 0 && self.class.associations?
           preload_associations(attributes, params, self.class)
+
+          # Handle assocs with no entries
+          params[:links].to_s.split(/,/).each do |name|
+            name, type, options = get_association(name,self.class)
+            instantiate_association(name, type)
+          end
         end
       end
 
       private
+        # Called to kick off the preloading of association objects from the response hash
+        #
         def preload_associations(attributes, params, scope)
           self.instance_exec(scope, attributes, &scope.preload_association)
         end
 
+        # Called by the subclass to instantiate an association object
+        #
         def add_association_object(scope, name, repr)
-          (type, options) = scope.associations[name.to_sym]
-          (type, options) = scope.associations[name.pluralize.to_sym] unless type
-          (type, options) = scope.associations[name.singularize.to_sym] unless type
-
+          name, type, options = get_association(name,scope)
           if type
             target = scope.model_class_for( options.fetch(:class_name) )
             result = target.instantiate_from(repr, params)
-            if is_collection?(type)
-              add_to_association_collection name, result
-            else
-              set_association_value name, result
-            end
+            set_or_add_association(name, type, result)
           else
             raise "Unhandled assocation: #{name}"
           end
         end
 
+        # Returns true if the association type is a collection
+        #
         def is_collection? type
           [ :polymorphic_belongs_to, :has_many ].include? type
         end
 
-        def add_to_association_collection name, target
-          variable = "@_#{name}"
-          instance_variable_set(variable, Collection.new) unless instance_variable_defined?(variable)
-          collection = instance_variable_get(variable)
-          target.respond_to?(:each) ? collection.concat(target) : collection.push(target)
+        # Returns the association metadata for the given association name. This
+        # method is partly needed because, while the association description
+        # may be pluralised, the includes() parameter will be singularised.
+        #
```

Comment:
And again, it means we should change Reticulum. In any case, the code that was replaced was doing exactly this.